### ISSUE TITLE
Add sorting options to Module Browser

### ISSUE THREAD

**Issue by AndrewBelt** [VCVRack/Rack#1434]

**Comment by Coirt** [comment]
Original, idea was the best!

**Comment by catronomix** [comment]
I do like this! Maybe among the sorting options, module width is an idea if possible :)

**Comment by zxol** [comment]
Sorting by recently used, as well as sorting by most used.

Letting the user tag each module would be awesome.  So you could tag "favorites" and a list of your favorite modules would come up, a bit like gmail/ google keep etc. 

Maybe a community popularity sorting index would be cool too.

Recently updated is another idea.

**Comment by gabrielreygoodlatte** [comment]
User tagging or favoriting of modules in the browser would be great. I guess favorites used to be a feature but I missed it.

**Comment by AndrewBelt** [comment]
@Coirt This issue is about sorting *options*, so **recently used** would be an example of such an option.

@catronomix I don't know if module width is possible because `ModelBox`es lazily create `ModuleWidget`s and doesn't know the width before it is created.

@zxol **most used** is possible. I can keep an enormous `modulePopularity` property in `settings.json`. **recently updated** is possible by reading the modification date of the plugin folder. Community data is a mess: It's time-variable, requires an internet connection, and new APIs, so I'm rejecting that idea.

@gabrielreygoodlatte I'm rejecting favorites, tagging, and rating at the moment, and anything else that requires manually setting data with a new UI.

**Comment by ducklin5** [comment]
@AndrewBelt I've had a look at the code and I can see that sorting by width would be tricky; it is still possible though. In order to get the width of all the modules, `model->createModuleWidgetNull()` will have to be called for each `model`. Of course, this would be terribly inefficient if it was run on every search. But what if it happens only once and the results are cached?

**Comment by XtremeSounds** [comment]
Yes I like this idea better.  the only issue is sometimes there are some really good modules that i forget about or certain modules i only want to use but forget their name etc.   How about instead of favorites we can add custom tags.  Would that help?
